sonar-db/src/main/java/org/sonar/db/permission/PermissionDao.java

```diff
@@ -98,6 +120,25 @@ public int countGroups(DbSession session, String permission, @Nullable Long comp
   }
 
   /**
+   * ordered by group names
+   */
+  public List<String> selectGroupNamesByPermissionQuery(DbSession dbSession, PermissionQuery query) {
+    return mapper(dbSession).selectGroupNamesByPermissionQuery(query, new RowBounds(query.getPageOffset(), query.getPageSize()));
+  }
+
+  public int countGroupsByPermissionQuery(DbSession dbSession, PermissionQuery query) {
+    return mapper(dbSession).countGroupsByPermissionQuery(query);
+  }
+
+  public List<GroupRoleDto> selectGroupPermissionsByQuery(DbSession dbSession, PermissionQuery query) {
+    if (query.getGroupNames() != null && query.getGroupNames().isEmpty()) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'getGroupNames' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)